webui/react/src/components/FilterForm/components/FilterField.tsx

```diff
@@ -204,7 +228,9 @@ const FilterField = ({
         <Select
           data-test="operator"
           options={(isSpecialColumn
-            ? [Operator.Eq, Operator.NotEq] // just Eq and NotEq for Special column
+            ? isSpecialMetadataColumn
+              ? [Operator.Contains, Operator.Eq, Operator.NotEq] // just Contain, Eq and NotEq for Special metadata column
+              : [Operator.Eq, Operator.NotEq] // just Eq and NotEq for Special column
             : AvailableOperators[currentColumn?.type ?? V1ColumnType.UNSPECIFIED]
```

Comment:
consider flattening `isSpecialColumn` and `isSpecialMetadataColumn` into an enum. We want to make sure that state that doesn't make sense is impossible to represent in the application. by using two booleans, we introduce the ability for `isSpecialColumn` to be false, but `isSpecialMetadataColumn` to be true. while we currently have runtime code guarding for that possibility, it can be hard to keep track of that invariant at code review. We should instead lift that requirement into the type system, so an issue with inexhaustive checks or new states will be caught in CI.

Reply:
fixed